lib/tests/streamlit/state/session_state_test.py

```diff
@@ -530,6 +532,19 @@ def test_maybe_set_state_value(self):
         assert self.session_state.maybe_set_state_value("widget_id_1") == True
         assert self.session_state["widget_id_1"] == 1
 
+        self.session_state._old_state["widget_id_1"] = 1
```

Comment:
nit: now that this test is getting more complicated, it'd be helpful to include an explanatory comment for each of the cases

alternatively (and maybe preferably), we could also split this into multiple tests

Reply:
I have split these out into separate cases. I think it's easier to understand as each test case is now separated and is not influenced by the other parts of the test (e.g.; the `del` lines are not needed).